cpp/src/gandiva/projector.cc

```diff
@@ -157,6 +181,7 @@ Status Projector::AllocArrayData(const DataTypePtr& type, int64_t num_records,
 
   // This is not strictly required but valgrind gets confused and detects this
   // as uninitialized memory access. See arrow::util::SetBitTo().
+  memset(null_bitmap->mutable_data(), 0, bitmap_bytes);
```

Comment:
this is a very frequent codepath. can we do the zeroing only when there is a selection vector ?